protostar/cheatable_starknet/controllers/expect_call_controller.py

```diff
@@ -49,6 +49,21 @@ def add_expected_call(self, expected_call: ExpectedCall):
                 expected_call
             ]
 
+    def assert_expect_call(self, expected_call: ExpectedCall):
+        contract_address = expected_call.address
+        expected_calls = self._cheatable_state.expected_contract_calls.get(
+            contract_address
+        )
+        if not expected_calls:
+            return
```

Comment:
This should raise the exception as well

Reply:
I don't think so, because it also means that there might be calls that were fulfilled - in this case `expected_calls` will also be empty because all the expected calls were removed. Also, if there are other expected calls that were not called, would we also have to raise an error? Example:
```
expect_call(a)
expect_call(b)
call(a)
assert_expect_call(a) // this is ok, a was called
assert_expect_call(c) // this should raise an error then
```
I don't think we should rise an error in such cases.

What `assert_expect_call` does is it checks that there's no hanging expected call that matches the specified criteria.

What you want `assert_expect_call` to do is: check if there were any expected calls with specified params that were not called.

To achieve that, I think we'd have to have two separate collections: `expected_calls` and `performed_calls`. This is because we'd have to check whether an expectation was made. I think this is a strong edge case and it brings some complexity.

Another thing is that it used to be implemented like that in the old cheat code.

I wouldn't go for it, what do you think?

Reply:
Ok let's discuss there: https://github.com/software-mansion/protostar/issues/1509#issuecomment-1441461319